cradle-cassandra/src/main/java/com/exactpro/cradle/cassandra/dao/messages/GroupedMessageIteratorProvider.java

```diff
@@ -92,7 +92,9 @@ public GroupedMessageIteratorProvider(String requestInfo,
 		this.returned = new AtomicInteger();
 		// TODO: Get message batch before *from* timestamp
 		this.leftBoundFilter = createLeftBoundFilter(filter);
+		this.firstPage = FilterUtils.findFirstPage(filter.getPageId(), leftBoundFilter, book);
```

Comment:
In this case we execute the FilterUtils.findFirstPage call twirly: here and in the createLeftBoundFilter method

Reply:
what do you suggest as an alternative ?